setup/src/main.ts

```diff
@@ -1,59 +1,201 @@
-import * as exec from '@actions/exec';
+import * as core from "@actions/core";
+import * as exec from "@actions/exec";
+import * as github from "@actions/github";
+import axios from "axios";
+import type { WebhookPayload } from "@actions/github/lib/interfaces";
 
+// commitsList recipe isn't here because the user of the action cannot define the list of commits
+const POSSIBLE_INPUT_RECIPES = [
+  "auto",
+  "uncommitted",
+  "mine",
+  "recent",
+  "all",
+  "lastPush",
+];
+
+interface Inputs {
+  sprklVersion: string;
+  analyze: string;
+  setEnv: string;
+  recipe: string;
+}
+//amount limit of commits to ask from the github api
+const PR_COMMITS_AMOUNT_LIMIT = 100;
+
+if (require.main === module) {
+  main();
+}
 
 /**
-    Setup sprkl function
+    Setup sprkl function.
  */
-(async () => {
-    // get the input values from the action
-    const sprklVersion = core.getInput('version');
-    const analyze = core.getInput('analyze');
-    const setEnv = core.getInput('setenv');
-
-    // run sprkl install command
-    const installCmd = `npx @sprkl/scripts@${sprklVersion} install`;
-    await exec.exec(installCmd);
-
-    // run sprkl analysis if requested
-    if (analyze === 'true') {
-        await exec.exec('sprkl apply');
-    }
+async function main() {
+  // get the input values from the action and make the InputsObj
+  const inputsObj: Inputs = {
+    sprklVersion: core.getInput("version"),
+    analyze: core.getInput("analyze"),
+    setEnv: core.getInput("setenv"),
+    recipe: core.getInput("recipe"),
+  };
+  const eventName = github.context.eventName;
+  // get the workflow json which include all the data about the workflow
+  const workflowPayload = github.context.payload;
 
-    // set sprkl environment if requested
-    if (setEnv === 'true') {
-        const sprklPrefix = await getSprklPrefixOrFail();
+  // validate the inputs from the action user(only analyze, setEnv and recipe. No vaildation for sprklVersion)
+  validateInputOrFail(inputsObj);
 
-        core.exportVariable('SPRKL_PREFIX', sprklPrefix);
-        core.exportVariable('NODE_OPTIONS','-r @sprkl/obs');
-        core.exportVariable('NODE_PATH', `${sprklPrefix}/lib/node_modules`)
+  // run sprkl install command
+  const installCmd = `npx @sprkl/scripts@${inputsObj.sprklVersion} install`;
+  await exec.exec(installCmd);
+
+  if (inputsObj.recipe === "auto") {
+    // get environment variables to set based on the event(SPRKL_RECIPE and more env vars based on recipe)
+    const envVarsToSet = await getRecipeEnv(eventName, workflowPayload);
+    // set all the environment variables in the recieved list
+    for (const [key, value] of envVarsToSet) {
+      core.exportVariable(key, value);
     }
-})();
+  } else {
+    // set sprkl recipe environment variable based on input
+    core.exportVariable("SPRKL_RECIPE", inputsObj.recipe);
+  }
+
+  // run sprkl analysis if requested
+  if (inputsObj.analyze === "true") {
+    await exec.exec("sprkl apply");
```

Comment:
@marom-sprkl add option to run in specific directory (which should be the repo where the checkout action placed the code)